scripts/prepare_disk.sh

```diff
@@ -196,6 +231,17 @@ if [ ! -r "${PARTITION_TABLE_FILE}" ]; then
     exit 1
 fi
 
+if [ ! -b "${TARGET_DISK}" ]; then
+    echo "Error, block device '${TARGET_DISK}' does not exist."
+    exit 1
+fi
+
+resize_needed="$(is_resize_needed; echo "${?}")"
+if [ "${resize_needed}" -eq 1 ]; then
```

Comment:
with the latest changes, can't this just be?
```
if ! is_resize_needed; then
```

Reply:
only if resize_need = /bin/true; otherwise it will try to execute '1' which of course doesn't exist

Reply:
Read again, I proposed to change BOTH lines. Just like how is_integer and is_comment are used.